Change log update (#1756)
microsoft/appcenter-sdk-apple: CHANGELOG.md

* **[Fix]** Fix crash involving SDK's `ms_viewWillAppear` method.

> ### App Center Data

Reviewer: Change 2.3.1 above to 2.4.0

Reviewer 2: Yep, breaking change on a preview module @MehrdadMzfr .

> * **[Breaking]** Rename the callback delegate method in `MSRemoteOperationDelegate` to `data:didCompleteRemoteOperation:forDocumentMetadata:withError:(MSDataError *_Nullable)error;`.

Reviewer: ```suggestion
* **[Breaking change]** Rename the callback delegate method in `MSRemoteOperationDelegate` to `data:didCompleteRemoteOperation:forDocumentMetadata:withError:(MSDataError *_Nullable)error;`.
```

> * **[Breaking change]** Rename the callback delegate method in `MSRemoteOperationDelegate` to `data:didCompleteRemoteOperation:forDocumentMetadata:withError:(MSDataError *_Nullable)error;`.

Reviewer 2: Would that be better: 
```suggestion
* **[Breaking change]** Rename delegate method `data:didCompletePendingOperation:forDocument:withError:` from `MSRemoteOperationDelegate` to `data:didCompleteRemoteOperation:forDocumentMetadata:withError:`.
```

> * **[Breaking change]** Rename delegate method `data:didCompletePendingOperation:forDocument:withError:` from `MSRemoteOperationDelegate` to `data:didCompleteRemoteOperation:forDocumentMetadata:withError:`.

Reviewer 3: I think it is okay but we haven't used `[Breaking change]`. Is it caused by a bug fix or additional features? Would be better to put more details why we change this **only if it is possible and reasonable**. Otherwise it should be good to keep it.

Author: @jaelim-ms 
the change is to match the Android callback, it's not an additional feature:
https://github.com/microsoft/appcenter-sdk-android/blob/4ae6533d2babfc50032e7aed30ab60a29ecb9361/sdk/appcenter-data/src/main/java/com/microsoft/appcenter/data/models/RemoteOperationListener.java#L22
